packages/ui/src/models/camel/camel-route-resource.ts

```diff
@@ -102,10 +107,19 @@ export class CamelRouteResource implements CamelResource, BeansAwareResource {
        * specialChildren is a map of processor names and their special children.
        */
       const specialChildren: Record<string, string[]> = {
-        choice: ['when', 'otherwise'],
-        doTry: ['doCatch', 'doFinally'],
+        choice: ['when'],
+        doTry: ['doCatch'],
       };
 
+      /** If an `otherwise` or a `doFinally` already exists, we shouldn't offer it in the catalog */
+      const definitionKeys = Object.keys(definition ?? {});
+      if (!definitionKeys.includes('otherwise')) {
+        specialChildren.choice.push('otherwise');
+      }
+      if (!definitionKeys.includes('doTry')) {
+        specialChildren.doTry.push('doTry');
```

Comment:
Shouldn't there be specified "doFinally"?

Reply:
You're right @tplevko, thanks for catching this up, I'll create another pull request to fix this.